File: 프로그래머스/1/340199. ［PCCE 기출문제］ 9번 ／ 지폐 접기/［PCCE 기출문제］ 9번 ／ 지폐 접기.java
class Solution {
    public int solution(int[] wallet, int[] bill) {
        int answer = 0;
        int width = Math.max(wallet[0], wallet[1]);
        int height = Math.min(wallet[0], wallet[1]);
        
        int big = Math.max(bill[0], bill[1]);
        int small = Math.min(bill[0], bill[1]);
        
        int temp = 0;
        
        while(true) {
            if(width >= big && height >= small) break;
            else {
                big /= 2;
                answer++;
                if(big < small) {
                    temp = small;
                    small = big;
                    big = temp;
                }
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/340199. ［PCCE 기출문제］ 9번 ／ 지폐 접기/［PCCE 기출문제］ 9번 ／ 지폐 접기.py
def solution(wallet, bill):
    answer = 0
    wallet.sort()
    while True:
        bill.sort() 
        print(bill)             
        if bill[1] > wallet[1]:
            bill[1] //=2
            answer+=1
            print(answer)
        elif bill[0] > wallet[0]:
            bill[1] //=2
            answer+=1
            print(answer)
        else: break

    return answer